.github/workflows/release.yaml

```
@@ -0,0 +1,58 @@
+name: Release
+
+on:
+  workflow_dispatch: {}
+
+jobs:
+  release:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+      with:
+        submodules: true
+        fetch-depth: 0
+    - name: Set environment variables
+      run: ./build/ci/set_environment.sh
+    - name: Build chart
+      run: ./build/chart.sh
+      env:
+        IMAGE_REPOSITORY: ghcr.io/${{ github.repository }}
+    - name: Draft release
+      id: draft_release
+      uses: actions/create-release@v1
+      with:
+        release_name: ${{ env.GIT_TAG }}
+        tag_name: ${{ env.GIT_TAG }}
+        body: A MITS release.
+        draft: true
+        prerelease: false
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+    - name: Upload chart to draft release
+      uses: actions/upload-release-asset@v1
+      with:
+        upload_url: ${{ steps.draft_release.outputs.upload_url }}
+        asset_path: output/mits-${{ env.VERSION }}.tgz
+        asset_name: mits-${{ env.VERSION }}.tgz
+        asset_content_type: application/x-gzip
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+    - name: Build and publish image
+      uses: docker/build-push-action@v1
+      with:
+        registry: ghcr.io
+        repository: ${{ github.repository }}
+        tags: ${{ env.IMAGE_TAG }}
+        labels: >-
+          org.opencontainers.image.version=${{ env.VERSION }}
+        add_git_labels: true
+        always_pull: true
+        dockerfile: image/Dockerfile
+        username: ${{ secrets.GHCR_USERNAME }}
+        password: ${{ secrets.GHCR_PASSWORD }}
```

```suggestion
        password: ${{ secrets.GITHUB_TOKEN }}
```
My understanding is that the Actions-provided token is magical: it means that pushes to GHCR will be free (whereas the personal token would be rate-limited once GHCR is out of beta).

---

Sounds good, doesn't work. Even the super-linter from GitHub doesn't use it.
https://github.com/github/super-linter/blob/daddeae7beda9b110508df6ebcbf035a0f66e14c/.github/workflows/deploy-RELEASE.yml#L63-L64

---

Ah, the [docs](https://docs.github.com/en/packages/getting-started-with-github-container-registry/migrating-to-github-container-registry-for-docker-images#updating-your-github-actions-workflow) says:
> GitHub Container Registry does not support using `GITHUB_TOKEN` for your PAT so you must use a different custom variable, such as `CR_PAT`.

That makes no sense, but it is what it is, so ¯\\\_(ツ)\_/¯